planning/behavior_path_planner/src/scene_module/avoidance/avoidance_module.cpp

```diff
@@ -1584,22 +1662,195 @@ double AvoidanceModule::getLeftShiftBound() const
+       // problem to extend for avoidance) lane if it exists
+      // this can be available only if line string is shared
+      const auto opt_right_lane = route_handler->getRightLanelet(lane);
+      const auto opt_left_lane = route_handler->getLeftLanelet(lane);
+
+      if (opt_right_lane) {
+        extended_lanelets.push_back(opt_right_lane.get());
+        continue;
+      } else if (opt_left_lane) {
+        extended_lanelets.push_back(opt_left_lane.get());
+        continue;
+      }
+    }
+
+    {  // 2. when there are multiple turning lanes whose previous lanelet is the same in
+       // intersection
+      const bool update_extended_lanelets = [&]() {
+        // lanelet is not turning lane
+        const std::string turn_direction = lane.attributeOr("turn_direction", "none");
+        if (turn_direction != "right" && turn_direction != "left") {
+          return false;
+        }
+
+        // get previous lane, and return false if previous lane does not exist
+        lanelet::ConstLanelet prev_lane;
+        if (!route_handler->getPreviousLaneletWithinRoute(lane, &prev_lane)) {
+          return false;
+        }
+
+        // get next lanes from the previous lane, and return false if next lanes do not exist
+        const auto next_lanes = route_handler->getNextLanelets(lane);
+        if (next_lanes.empty()) {
+          return false;
+        }
 
-  double right_extend = std::min(*right_extend_elem, 0.0);
-  double left_extend = std::max(*left_extend_elem, 0.0);
+        // look for neighbour lane, where end line of the lane is connected to end line of the
+        // original lane
+        for (const auto & next_lane : next_lanes) {
+          if (lane.id() == next_lane.id()) {
+            continue;
+          }
+
+          const Eigen::Vector2d & next_left_back_point_2d =
+            next_lane.leftBound2d().back().basicPoint();
+          const Eigen::Vector2d & next_right_back_point_2d =
+            next_lane.rightBound2d().back().basicPoint();
+
+          const Eigen::Vector2d & orig_left_back_point_2d = lane.leftBound2d().back().basicPoint();
+          const Eigen::Vector2d & orig_right_back_point_2d =
+            lane.rightBound2d().back().basicPoint();
+
+          constexpr double epsilon = 1e-5;
+          const bool is_neighbour_lane =
+            (next_left_back_point_2d - orig_right_back_point_2d).norm() < epsilon ||
+            (next_right_back_point_2d - orig_left_back_point_2d).norm() < epsilon;
+          if (is_neighbour_lane) {
+            extended_lanelets.push_back(next_lane);
+            return true;
+          }
+        }
 
-  constexpr double THRESHOLD = 0.01;
-  right_extend -= (right_extend < -THRESHOLD) ? margin : 0.0;
-  left_extend += (left_extend > THRESHOLD) ? margin : 0.0;
+        return false;
+      }();
+      if (update_extended_lanelets) {
+        continue;
+      }
+    }
 
-  const auto extended_lanelets = lanelet::utils::getExpandedLanelets(
```

Comment:
@zulfaqar-azmi-t4 
Note:
removed extend lanelets by constant margin

Reply:
Yes, this removes [fix lanelet expansion](https://github.com/autowarefoundation/autoware.universe/pull/287/files#:~:text=constexpr%20double%20extend_margin%20%3D%200.5%3B) in favor of directly insert the lanelet to 
extended drivable areas list.